src/Entity/WSCall.php

```diff
@@ -97,7 +100,48 @@ public function getLanguagePlugin() {}
   /**
    * {@inheritdoc}
    */
-  public function call($type, $key = NULL, $replacement = [], $argument = [], $options = [], &$method = '') {}
+  public function call($method = NULL, $replacements = [], $data = NULL, $options = [], $key = NULL, $tokens = [])  {
+    // Build out the Cache ID based on the parameters passed.
+    $cid_array = array_merge($options, $this->getOptions(), $replacements, $tokens, array('data' => $data, 'key' => $key));
+    $cid = md5(serialize($cid_array));
+    if ($cache = \Drupal::cache('wsdata')->get($cid)) {
+      $cache_data = $cache->data;
```

Comment:
Instead of wrapping the entire contents of this function in an "else", just return early if you have the data in the cache.

Reply:
I added the return early.